src/deploy/functions/prepareFunctionsUpload.ts

```diff
@@ -102,13 +112,14 @@ async function packageSource(
       filesize(archive.pointer()) +
       ") for uploading"
   );
-  return tmpFile;
+  const hash = fileHashes.join(".");
```

Comment:
woah isn't this hash like really long? One hash per every file joined via "."?

Reply:
🙃 At this point yeah its a super long string. By the time we get to storing it on the endpoint we will sha1 the long string into a shorter string.

An alternative is to pass around something like: `interface FileHash  { filename: string; hash: string}`.

WDYT?

Reply:
hmm I thought you'd create an "intermediate" hash in this function:

```js
  const hash = crypto.createHash("sha1"); // or md5 or whatever
...
  hash.update(fileHash);
```

Hash of a hash of a hash of a hash of a...

To think about it - probably doesn't matter too much whether we carry around a hash or some really long string 🤷🏼.

Reply:
What gets passed around doesn't matter too much, as long as the final hash we store as a label isn't large. 🤷